SettingsList.py

```diff
@@ -3005,15 +3005,57 @@ def __init__(self, name, gui_text, min, max, default, step=1,
             'option_remove': ['fortress'],
         },
     ),
+    Combobox(
+        name           = 'shuffle_tcgkeys',
+        gui_text       = 'Treasure Chest Game Keys',
+        default        = 'vanilla',
+        choices        = {
+            'remove':      "Remove (Keysy)",
+            'vanilla':     "Vanilla Locations",
+            'regional':    "Regional",
+            'overworld':   "Overworld Only",
+            'any_dungeon': "Any Dungeon",
+            'keysanity':   "Anywhere (Keysanity)",
+        },
+        gui_tooltip    = '''\
+            'Remove': All Treasure Chest Game Keys will be removed
+            and all doors will remained unlocked.
+
+            'Vanilla': Treasure Chest Game Keys will have vanilla
+            behavior (one random per room). The minigame will
+            also have vanilla behavior.
+
+            'Regional': Treasure Chest Game Keys can only appear
+            in regions near the minigame (including
+            the minigame itself or other dungeons in the region, such as Ganon's Castle).
+            <a href="https://wiki.ootrandomizer.com/index.php?title=Hints#Hint_Regions" target="_blank">The Wiki has a list of corresponding regions here.</a>
+
+            'Overworld Only': Treasure Chest Game Keys can only appear
+            outside of dungeons. You may need to enter the Treasure Chest Game
+            multiple times to gain items to access the overworld locations
+            with the keys required to clear the minigame.
+
+            'Any Dungeon': Treasure Chest Game Keys can only appear inside of a dungeon.
+            A difficult mode since it is more likely to need to enter the minigame
+            multiples times.
+
+            'Anywhere': Treasure Chest Game Keys can appear anywhere
+            in the world.
+        ''',
+        shared         = True,
+        gui_params     = {
+            'randomize_key': 'randomize_settings',
+        },
+    ),
     Combobox(
         name           = 'key_rings_choice',
         gui_text       = 'Key Rings Mode',
         default        = 'off',
         choices        = {
             'off':       'Off',
-            'choice':    'Choose dungeons',
-            'all':       'All dungeons',
-            'random':    'Random dungeons'
+            'choice':    'Choose dungeons/Treasure Chest Game',
```

Comment:
If you're going to change this, also add Thieves' Hideout here. I'd say it's not necessary in the option names and should probably be mentioned in the tooltip instead.

Reply:
addressed